### DISCUSSION TITLE
openshift installer vSphere and API/VIPS? No guide on VIP setup

### DISCUSSION THREAD

**Discussion by raymond-itsec** [okd-project/okd#1524] [Q&A]
Environment: vSphere 7.0.2
Knowledge level OKD: newbie

I am new to OKD and tried to setup OKD using openshift-installer create cluster (docs on https://docs.okd.io/latest/installing/installing_vsphere/preparing-to-install-on-vsphere.html) 

vSphere does create a bootstrap node and 3 master nodes with FCOS. 

However the API/VIP addresses do not get assigned to a system/bootstrap node/pod. So is there a guide that works with the vSphere API to create everything using yaml scripts?

**Comment by vrutkovs** [comment]
>However the API/VIP addresses do not get assigned to a system/bootstrap node/pod

Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting) to find out more

> So is there a guide that works with the vSphere API to create everything using yaml scripts

We have multiple of them:
* [govc](https://github.com/okd-project/okd/tree/master/Guides/UPI/vSphere_govc)
* [terraform](https://github.com/okd-project/okd/tree/master/Guides/UPI/vSphere_terraform)
* and [docs](https://docs.okd.io/latest/installing/installing_vsphere/installing-vsphere.html) on the whole custom flow
